pkg/kubectl/cmd/diff.go

```diff
@@ -216,82 +215,55 @@ func (d *Directory) Delete() error {
 // Object is an interface that let's you retrieve multiple version of
 // it.
 type Object interface {
-	Live() (map[string]interface{}, error)
-	Merged() (map[string]interface{}, error)
+	Live() runtime.Object
+	Merged() (runtime.Object, error)
 
 	Name() string
 }
 
 // InfoObject is an implementation of the Object interface. It gets all
 // the information from the Info object.
 type InfoObject struct {
-	Remote  *unstructured.Unstructured
-	Info    *resource.Info
-	Encoder runtime.Encoder
-	Parser  *parse.Factory
+	LocalObj runtime.Object
+	Info     *resource.Info
+	Encoder  runtime.Encoder
+	OpenAPI  openapi.Resources
 }
 
 var _ Object = &InfoObject{}
 
-func (obj InfoObject) toMap(data []byte) (map[string]interface{}, error) {
-	m := map[string]interface{}{}
-	if len(data) == 0 {
-		return m, nil
-	}
-	err := json.Unmarshal(data, &m)
-	return m, err
+func (obj InfoObject) Live() runtime.Object {
+	return obj.Info.Object
 }
 
-func (obj InfoObject) Live() (map[string]interface{}, error) {
-	if obj.Remote == nil {
-		return nil, nil // Object doesn't exist on cluster.
+func (obj InfoObject) Merged() (runtime.Object, error) {
+	// Build the patcher, and then apply the patch with dry-run, unless the object doesn't exist, in which case we need to create it.
+	if obj.Live() == nil {
+		// Dry-run create if the object doesn't exist.
+		return resource.NewHelper(obj.Info.Client, obj.Info.Mapping).Create(
+			obj.Info.Namespace,
+			true,
+			obj.LocalObj,
+			&metav1.CreateOptions{DryRun: []string{metav1.DryRunAll}},
+		)
 	}
-	return obj.Remote.UnstructuredContent(), nil
-}
 
-func (obj InfoObject) Merged() (map[string]interface{}, error) {
-	data, err := runtime.Encode(obj.Encoder, obj.Info.Object)
+	modified, err := kubectl.GetModifiedConfiguration(obj.LocalObj, true, unstructured.UnstructuredJSONScheme)
```

Comment:
Should we use "false" here so that the change to the annotation doesn't appear in the diff? I don't know if that's useful.

Reply:
Yes. I don't think we're interested in seeing the difference of the annotation--only the fields.